### PR TITLE
Golang poseidon docs

### PR DESCRIPTION
**Pull request by jeremyfelder**
## Describe the changes

This PR adds docs for poseidon in golang bindings

### CODE REVIEW

File: docs/docs/icicle/golang-bindings/hash.md [hunk 1]
```diff
@@ -72,3 +72,37 @@ fmt.Println!("Hash(`", input_str, "`) =", &outputAsHexStr)
...
+
+Since poseidon is designed for use with field elements and curves, it is located within the field or curve packages and not in the Hash package though it does rely on using the Hash package.
+
+:::
+
+```golang
+import (
+  "github.com/ingonyama-zk/icicle/v3/wrappers/golang/core"
+  babybear "github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear/poseidon"
```
Comment (LeonHibnik) [inline docs/docs/icicle/golang-bindings/hash.md:90]:
```suggestion
  "github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear/poseidon"
```

File: docs/docs/icicle/golang-bindings/hash.md [hunk 2]
```diff
@@ -72,3 +72,37 @@ fmt.Println!("Hash(`", input_str, "`) =", &outputAsHexStr)
...
+
+The Poseidon hash is designed for cryptographic field elements and curves, making it ideal for use cases such as zero-knowledge proofs (ZKPs). Poseidon hash using babybear field:
+
+:::tip
+
+Since poseidon is designed for use with field elements and curves, it is located within the field or curve packages and not in the Hash package though it does rely on using the Hash package.
+
+:::
+
+```golang
```
Comment (LeonHibnik) [inline docs/docs/icicle/golang-bindings/hash.md:86]:
```suggestion
```go
```

File: docs/docs/icicle/golang-bindings/hash.md [hunk 3]
```diff
@@ -72,3 +72,37 @@ fmt.Println!("Hash(`", input_str, "`) =", &outputAsHexStr)
...
+import (
+  "github.com/ingonyama-zk/icicle/v3/wrappers/golang/core"
+  babybear "github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear/poseidon"
+)
+
+	batch := 1 << 4
```
Comment (LeonHibnik) [inline docs/docs/icicle/golang-bindings/hash.md:93]: Remove tabs and leading spaces

File: docs/docs/icicle/golang-bindings/hash.md [hunk 4]
```diff
@@ -72,3 +72,37 @@ fmt.Println!("Hash(`", input_str, "`) =", &outputAsHexStr)
 ```
 
 Using other hash algorithms is similar and only requires replacing the Hasher constructor with the relevant hashing algorithm.
+
+### 3. Poseidon Example (field elements) and batch hashing
+
+The Poseidon hash is designed for cryptographic field elements and curves, making it ideal for use cases such as zero-knowledge proofs (ZKPs). Poseidon hash using babybear field:
+
+:::tip
```
Comment (LeonHibnik) [inline docs/docs/icicle/golang-bindings/hash.md:80]: Why tip and not note or info tag?

**Review by LeonHibnik — COMMENTED** [review]
Left minor comments, looks good overall!

**Review by LeonHibnik — APPROVED** [review]

File: docs/docs/icicle/golang-bindings/hash.md [hunk 5]
```diff
@@ -72,3 +72,37 @@ fmt.Println!("Hash(`", input_str, "`) =", &outputAsHexStr)
...
+```go
+import (
+  "github.com/ingonyama-zk/icicle/v3/wrappers/golang/core"
+  babybear "github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear"
+  "github.com/ingonyama-zk/icicle/v3/wrappers/golang/fields/babybear/poseidon"
+)
+
+batch := 1 << 4
+t := 3 // Currently support arity of 3, 5, 9, 12
+inputsSize = batch * (t - 1)
```
Comment (yshekel) [inline docs/docs/icicle/golang-bindings/hash.md:95]: maybe worth adding a comment that 't-1' is due to use of domain-tag

**Review by yshekel — APPROVED** [review]

**Pull request merged**